Sources/Apollo/RecordSet.swift

```diff
@@ -14,6 +14,14 @@ public struct RecordSet {
     storage.removeValue(forKey: key)
   }
 
+  public mutating func removeRecords(matching pattern: CacheKey) {
+    for (key, _) in storage {
+      if key.contains(pattern) {
```

Comment:
`contains` seemed like the most efficient choice here given that deletion can happen within. It's unfortunate that `Dictionary` doesn't support something like `removeAll(where:)` as `Array` does.